website/docs/concepts/router.md

```diff
@@ -31,65 +39,18 @@ enum Route {
 }
 ```
 
-A `Route` is paired with a `<Switch />` component, which finds the first variant whose path matches the
-browser's current URL and passes it to the `render` callback. The callback then decides what to render.
-In case no path is matched, the router navigates to the path with `not_found` attribute. If no route is specified,
-nothing is rendered, and a message is logged to console stating that no route was matched.
-
-```rust
-use yew_router::prelude::*;;
-use yew::prelude::*;
-
-#[derive(Clone, Routable, PartialEq)]
-enum Route {
-    #[at("/")]
-    Home,
-    #[at("/secure")]
-    Secure,
-    #[not_found]
-    #[at("/404")]
-    NotFound,
-}
-
-#[function_component(Secure)]
-fn secure() -> Html {
-    let history = use_history().unwrap();
-
-    let onclick_callback = Callback::from(move |_| history.push(Route::Home));
-    html! {
-        <div>
-            <h1>{ "Secure" }</h1>
-            <button onclick={onclick_callback}>{ "Go Home" }</button>
-        </div>
-    }
-}
-
-fn switch(routes: &Route) -> Html {
-    match routes {
-        Route::Home => html! { <h1>{ "Home" }</h1> },
-        Route::Secure => html! {
-            <Secure />
-        },
-        Route::NotFound => html! { <h1>{ "404" }</h1> },
-    }
-}
-
-#[function_component(Main)]
-fn app() -> Html {
-    html! {
-        <Switch<Route> render={Switch::render(switch)} />
-    }
-}
-```
+A `Route` is paired with a `<Switch />` component, which finds the first variant whose path matches the browser's
```

Comment:
This is incorrect. [This test](https://github.com/yewstack/yew/blob/master/packages/yew-router/tests/router.rs) also checks the order (see how `/`, which should match everything, is at the top)

Reply:
sorry I don't get what you mean? How should this be corrected?

Reply:
> which finds the first variant whose path matches the browser's [...]

It is order *in*dependent. The variant with the required path is matched or it 404s. The test case checks this by keeping `/` as the first variant so if any other variants are matched, it means order doesn't matter.

Reply:
fixed